ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -38,29 +80,89 @@
    */
   GetManager(ClusterMap clusterMap, ResponseHandler responseHandler, RouterConfig routerConfig,
       NonBlockingRouterMetrics routerMetrics, Time time) {
-    //@todo
+    this.clusterMap = clusterMap;
+    this.responseHandler = responseHandler;
+    this.routerConfig = routerConfig;
+    this.routerMetrics = routerMetrics;
+    this.time = time;
+    getOperations = Collections.newSetFromMap(new ConcurrentHashMap<GetOperation, Boolean>());
   }
 
-  void submitGetBlobOperation(String blobId, FutureResult<ReadableStreamChannel> futureResult,
-      Callback<ReadableStreamChannel> callback) {
-    //@todo
+  /**
+   * Submit an operation to get the BlobInfo associated with a blob asynchronously.
+   * @param blobId the blobId for which the BlobInfo is being requested, in string form.
+   * @param futureResult the {@link FutureResult} that contains the pending result of the operation.
```

Comment:
that is how it is used elsewhere as well, as the result is pending at the time of submission.